src/commands/synthetics/batch.ts

```diff
@@ -268,6 +274,31 @@ const getResultFromBatch = (
   }
 }
 
+const getResultWithoutPollResult = (
```

Comment:
Given the similarity between this function and the return statement from the above, could we merge the two, and add only the difference if `pollResult` is not available?
```typescript
const resultFromBatch = {
    executionRule: resultInBatch.execution_rule,
    initialResultId: resultInBatch.initial_result_id,
    isNonFinal: isNonFinalResult(resultInBatch),
    location: getLocation(resultInBatch.location, test),
    passed: hasResultPassed(resultInBatch, isUnhealthy, hasTimedOut, options),
    resultId: getResultIdOrLinkedResultId(resultInBatch),
    retries: resultInBatch.retries || 0,
    maxRetries: resultInBatch.max_retries || 0,
    selectiveRerun: resultInBatch.selective_rerun,
    timedOut: hasTimedOut,
}

if (pollResult) {
  return {
    ...resultFromBatch,
    result: {
      ...pollResult.result,
      ...(safeDeadlineReached ? {
        failure: new BatchTimeoutRunawayError().toJson()
        passed: false
        } : timedOutRetry || hasTimedOut ? {
          failure: new {code: 'TIMEOUT', message: 'The batch timed out before receiving the result.'}
          passed: false
        } : {})
    },
    test: deepExtend({}, test, pollResult),
    timestamp: pollResult.timestamp
  }
} else {
  return {
    ...resultFromBatch,
    test: deepExtend({}, test),
    timestamp: Date.now()
  }
}
```
This code might not be the best solution, but will hopefully give you an idea.

Reply:
This way would make it more concise, but i'm not sure if having the two functions is not actually clearer and easier to understand 🤔

Reply:
I prefer the 2 functions as well tbh 😁